### ISSUE TITLE
Experiencing several Different Errors importing to PT

### ISSUE THREAD

**Issue by benstuart0** [evolutionsoftswiss/pokernow-handhistory-converter#3]
Any idea why PokerTracker isn't accepting the converted format?

```
06:44:55 pm: Error: PokerStars: Encountered unexpected end of hand history. (Hand #20210720025730218) (Line #7907)
06:44:55 pm: Error: PokerStars: [2] Invalid pot size for hand #20210720025730219 (12.00 vs 10.00) (Line #7948)
06:44:55 pm: Error: PokerStars: Encountered unexpected end of hand history. (Hand #20210720025730221) (Line #7982)
06:44:55 pm: Error: PokerStars: Encountered unexpected end of hand history. (Hand #20210720025730222) (Line #8003)
06:44:55 pm: Error: PokerStars: [2] Invalid pot size for hand #20210720025730223 (12.00 vs 10.00) (Line #8049)
06:44:55 pm: Error: PokerStars: [2] Invalid pot size for hand #20210720025730224 (27.00 vs 4.00) (Line #8071)
06:44:55 pm: Error: PokerStars: Encountered unexpected end of hand history. (Hand #20210720025730225) (Line #8082)
06:44:55 pm: Error: PokerStars: Encountered unexpected end of hand history. (Hand #20210720025730226) (Line #8103)
06:44:55 pm: Error: PokerStars: [2] Invalid pot size for hand #20210720025730227 (33.00 vs 0.00) (Line #8148)
06:44:55 pm: Error: PokerStars: Kanye2020 action out of sequence (Hand #20210720025730228) (Line #8175)
06:44:55 pm: Error: PokerStars: Encountered unexpected end of hand history. (Hand #20210720025730229) (Line #8186)
06:44:55 pm: Error: PokerStars: Encountered unexpected end of hand history. (Hand #20210720025730230) (Line #8207)
06:44:55 pm: Error: PokerStars: [2] Invalid pot size for hand #20210720025730231 (12.00 vs 0.00) (Line #8240)
06:44:55 pm: Error: PokerStars: Kanye2020 action out of sequence (Hand #20210720025730232) (Line #8271)
06:44:55 pm: Error: PokerStars: Encountered unexpected end of hand history. (Hand #20210720025730233) (Line #8282)
06:44:55 pm: Error: PokerStars: Encountered unexpected end of hand history. (Hand #20210720025730234) (Line #8312)
06:44:55 pm: Error: PokerStars: [2] Invalid pot size for hand #20210720025730235 (12.00 vs 0.00) (Line #8344)
06:44:55 pm: Error: PokerStars: Encountered unexpected end of hand history. (Hand #20210720025730239) (Line #8464)
06:44:55 pm: Error: PokerStars: Encountered unexpected end of hand history. (Hand #20210720025730240) (Line #8501)
06:44:55 pm: Error: PokerStars: [2] Invalid pot size for hand #20210720025730241 (12.00 vs 10.00) (Line #8549)
06:44:55 pm: Error: PokerStars: Invalid pot size (0.00 vs pot: 12.00 rake: 0.00 jpt: 0.00) for hand #20210720025730243 (Line #8622)
06:44:55 pm: Error: PokerStars: [2] Invalid pot size for hand #20210720025730245 (154.00 vs 64.00) (Line #8699)
06:44:55 pm: Error: PokerStars: Encountered unexpected end of hand history. (Hand #20210720025730246) (Line #8713)
06:44:55 pm: Error: PokerStars: Encountered unexpected end of hand history. (Hand #20210720025730247) (Line #8758)
06:44:55 pm: Error: PokerStars: [2] Invalid pot size for hand #20210720025730248 (35.00 vs 10.00) (Line #8823)
06:44:55 pm: Error: PokerStars: Invalid pot size (0.00 vs pot: 12.00 rake: 0.00 jpt: 0.00) for hand #20210720025730250 (Line #8894)
06:44:55 pm: Error: PokerStars: [2] Invalid pot size for hand #20210720025730252 (12.00 vs 4.00) (Line #8956)
06:44:55 pm: Error: PokerStars: Encountered unexpected end of hand history. (Hand #20210720025730253) (Line #8970)
06:44:55 pm: Error: PokerStars: Encountered unexpected end of hand history. (Hand #20210720025730254) (Line #9002)
06:44:55 pm: Error: PokerStars: [2] Invalid pot size for hand #20210720025730255 (12.00 vs 10.00) (Line #9050)
06:44:55 pm: Error: PokerStars: latvia action out of sequence (Hand #20210720025730256) (Line #9094)
06:44:55 pm: Error: PokerStars: [2] Invalid pot size for hand #20210720025730257 (31.00 vs 12.00) (Line #9124)
```

**Comment by evolutionsoftswiss** [comment]
Thanks for the feedback, Ben.
Did the issue appeared on a recently downloaded hand history ?
There may have been a change in the pokernow format causing the problems. It is possible that the code needs an update.
Could you attach the original csv file ?
Or attach a part of it after having sorted the rows by the third column order ?

**Comment by benstuart0** [comment]
Yes it's recent. I think you're right that PokerNow changed their formatting, all other available hand converters don't work as well. I'm attaching one of the original files.

[poker_now_log_lXarbUfwldTz_vbdjYmCZC69X.csv](https://github.com/evolutionsoftswiss/pokernow-handhistory-converter/files/6859227/poker_now_log_lXarbUfwldTz_vbdjYmCZC69X.csv)

**Comment by evolutionsoftswiss** [comment]
I could not reproduce the error messages with the provided csv. 100 Hands from there got converted and are importable without errors in my PT4. I've attached the converted txt file. Did  the error messages appear on another csv file ?

[poker_now_log_lXarbUfwldTz_vbdjYmCZC69X-converted.txt](https://github.com/evolutionsoftswiss/pokernow-handhistory-converter/files/6862018/poker_now_log_lXarbUfwldTz_vbdjYmCZC69X-converted.txt)

**Comment by benstuart0** [comment]
Hmm. It seemed every hand had an error when I tried to import.

What version of poker tracker do you use? What version of your software do you use? & are there any import configurations that I'm missing on PT?

**Comment by evolutionsoftswiss** [comment]
I've used v0.2.0 of pokernow-handhistory-converter with two or three additional log statements not part of the release. Poker Tracker I'm excluding as source for the different behavior.
In the meantime I suppose the different behavior comes from the operating system. I've tested on Windows only and the line endings on a Mac could cause log messages from poker tracker about unexpected hand endings. Are you importing an a Mac OS ?
Does replacing the present line endings '\n' with '\r' in the converted file resolve the import errors ?

**Comment by benstuart0** [comment]
Yes I'm using a mac, that would make sense, although it doesn't necessarily explain the pot size errors. I can try that.

**Comment by evolutionsoftswiss** [comment]
A new version of the converter program should avoid non-Mac conform line separators, you could also try with this one, it is not released yet: 
[handhistory-converter-0.2.1-program.zip](https://github.com/evolutionsoftswiss/pokernow-handhistory-converter/files/6866686/handhistory-converter-0.2.1-program.zip)

**Comment by benstuart0** [comment]
There's another bug on running this version. Needed source code with last version to work past it. I opened a new issue for that bug. I hope me identifying bugs is helpful and not annoying, don't try and fix things for my sake–it seems the issues are specific to Mac. Thanks for working through this with me!

**Comment by evolutionsoftswiss** [comment]
No worry Ben. Your feedback is helpful. I think you also found a general bug with the $ sign in the other issue.

**Comment by evolutionsoftswiss** [comment]
That will get difficult to track down. The line endings were changed to Mac style and the hands are still not imported.
I'm using PokerTracker Version 4.15.29 and all Hands with the line endings from Mac get imported also on my Windows.
I attach my conversion file again, but unfortunately I can't do a real test on a Mac. Maybe there are differences between the poker tracker versions or its behavior on different systems ?
[poker_now_log_lXarbUfwldTz_vbdjYmCZC69X-convertedMacLineEndings.txt](https://github.com/evolutionsoftswiss/pokernow-handhistory-converter/files/6884936/poker_now_log_lXarbUfwldTz_vbdjYmCZC69X-convertedMacLineEndings.txt)

**Comment by benstuart0** [comment]
I'll look into it a bit more on my system tomorrow and let you know what I find.

On Jul 27, 2021, at 4:18 AM, evolutionsoftswiss ***@***.***> wrote:

﻿
Non-NU Email
________________________________


That will get difficult to track down. The line endings were changed to Mac style and the hands are still not imported.
I'm using PokerTracker Version 4.15.29 and all Hands with the line endings from Mac get imported also on my Windows.
I attach my conversion file again, but unfortunately I can't do a real test on a Mac. Maybe there are differences between the poker tracker versions or its behavior on different systems ?
poker_now_log_lXarbUfwldTz_vbdjYmCZC69X-convertedMacLineEndings.txt<https://urldefense.com/v3/__https://github.com/evolutionsoftswiss/pokernow-handhistory-converter/files/6884936/poker_now_log_lXarbUfwldTz_vbdjYmCZC69X-convertedMacLineEndings.txt__;!!PvXuogZ4sRB2p-tU!XPoewYHwJ4_cIKNrDD3Cl7D4xdpqLOqc8Q_s37inMCFxldPW_jTCEjlAj39JycXg9s0qcmU$>

**Comment by benstuart0** [comment]
Using the file you attached in your last response, there were no import errors. However, I'm unable to replicate that. I've tried a few methods of changing to DOS line endings on my converted text files but it doesn't seem to work.

**Comment by benstuart0** [comment]
For some reason, the specific file I sent to you works even without converting line endings. There are only 100 hands in "poker_now_log_lXarbUfwldTz_vbdjYmCZC69X.txt" and all have no error. I'll attach another file that doesn't work, even when I've attempted to change line endings. About half of the hands import successfully, but the other half has import errors, mostly relating to pot size.

Note: The successful file is a Pot Limit Omaha game, while the unsuccessful files are NLH.

[poker_now_log_aSr5GallGPyC08-sFSpht0AwC.csv](https://github.com/evolutionsoftswiss/pokernow-handhistory-converter/files/6896968/poker_now_log_aSr5GallGPyC08-sFSpht0AwC.csv)

**Comment by evolutionsoftswiss** [comment]
Thanks for the additional history csv. I'll look at the converted NLH hands. Your hints are a good starting point.

**Comment by evolutionsoftswiss** [comment]
I think the existing problems were caused by player names with spaces. What are the results with this version ? Could not test it well, because of my Omaha only PokerTracker...
[handhistory-converter-0.2.3-program.zip](https://github.com/evolutionsoftswiss/pokernow-handhistory-converter/files/6897986/handhistory-converter-0.2.3-program.zip)

**Comment by benstuart0** [comment]
Unfortunately, still no change in errors.

> 01:54:25 am: Import file: /Users/Ben/Documents/PokerNow/convertedPokerNowHands/poker_now_log_aSr5GallGPyC08-sFSpht0AwC-converted.txt
01:54:25 am: Error: PokerStars: Encountered unexpected end of hand history. (Hand #2021072020500940) (Line #1412)
01:54:25 am: Error: PokerStars: Invalid pot size (105.00 vs pot: 104.00 rake: 0.00 jpt: 0.00) for hand #2021072020500941 (Line #1480)
01:54:25 am: Error: PokerStars: [2] Invalid pot size for hand #2021072020500943 (11.00 vs 7.00) (Line #1553)
01:54:25 am: Error: PokerStars: Conner action out of sequence (Hand #2021072020500944) (Line #1595)
01:54:25 am: Error: PokerStars: ZayZay action out of sequence (Hand #2021072020500947) (Line #1734)
01:54:25 am: Error: PokerStars: Encountered unexpected end of hand history. (Hand #2021072020500948) (Line #1751)
01:54:25 am: Error: PokerStars: [2] Invalid pot size for hand #2021072020500949 (2.00 vs 1.00) (Line #1823)
01:54:25 am: Error: PokerStars: [2] Invalid pot size for hand #2021072020500955 (55.00 vs 5.00) (Line #2101)
01:54:25 am: Error: PokerStars: Encountered unexpected end of hand history. (Hand #2021072020500957) (Line #2155)
01:54:25 am: Error: PokerStars: [2] Invalid pot size for hand #2021072020500958 (5.00 vs 4.00) (Line #2220)
01:54:25 am: Error: PokerStars: Encountered unexpected end of hand history. (Hand #2021072020500966) (Line #2546)
01:54:25 am: Error: PokerStars: [2] Invalid pot size for hand #2021072020500967 (39.00 vs 38.00) (Line #2613)
01:54:25 am: Error: PokerStars: [2] Invalid pot size for hand #2021072020500971 (15.00 vs 5.00) (Line #2787)
01:54:25 am: Error: PokerStars: Ciivil action out of sequence (Hand #2021072020500973) (Line #2884)
01:54:25 am: Error: PokerStars: [2] Invalid pot size for hand #2021072020500974 (11.00 vs 7.00) (Line #2923)
01:54:25 am: Error: PokerStars: Encountered unexpected end of hand history. (Hand #2021072020500975) (Line #2939)
01:54:25 am: Error: PokerStars: [2] Invalid pot size for hand #2021072020500976 (102.00 vs 4.00) (Line #3019)
01:54:25 am: Error: PokerStars: [2] Invalid pot size for hand #2021072020500978 (81.00 vs 13.00) (Line #3107)
01:54:25 am: Error: PokerStars: Encountered unexpected end of hand history. (Hand #2021072020500983) (Line #3295)
01:54:25 am: Error: PokerStars: [2] Invalid pot size for hand #2021072020500984 (39.00 vs 38.00) (Line #3367)
01:54:25 am: Error: PokerStars: [2] Invalid pot size for hand #2021072020500986 (5.00 vs 2.00) (Line #3450)
01:54:25 am: Error: PokerStars: [2] Invalid pot size for hand #2021072020500990 (27.00 vs 21.00) (Line #3633)
01:54:26 am: Adjusting session stakes ...
01:54:38 am: Updating the cache ...
01:54:39 am: 		+ Complete (289 hands, 0 summaries, 22 errors, 0 duplicates)
01:54:39 am: Import file: /Users/Ben/Documents/PokerNow/convertedPokerNowHands/poker_now_log_-6QvGr9-dySJZTL9yvTJiMhZZ-converted.txt
01:54:39 am: Error: PokerStars: Encountered unexpected end of hand history. (Hand #2021072002573024) (Line #681)
01:54:39 am: Error: PokerStars: Invalid pot size (321.00 vs pot: 319.00 rake: 0.00 jpt: 0.00) for hand #2021072002573025 (Line #756)
01:54:39 am: Error: PokerStars: [2] Invalid pot size for hand #2021072002573029 (67.00 vs 52.00) (Line #913)
01:54:39 am: Error: PokerStars: [2] Invalid pot size for hand #2021072002573030 (12.00 vs 4.00) (Line #948)
01:54:39 am: Error: PokerStars: Encountered unexpected end of hand history. (Hand #2021072002573032) (Line #1006)
01:54:39 am: Error: PokerStars: [2] Invalid pot size for hand #2021072002573033 (4.00 vs 2.00) (Line #1066)
01:54:39 am: Error: PokerStars: [2] Invalid pot size for hand #2021072002573035 (47.00 vs 32.00) (Line #1139)
01:54:39 am: Error: PokerStars: [2] Invalid pot size for hand #2021072002573037 (37.00 vs 12.00) (Line #1215)
01:54:39 am: Error: PokerStars: Encountered unexpected end of hand history. (Hand #2021072002573040) (Line #1308)
01:54:39 am: Error: PokerStars: [2] Invalid pot size for hand #2021072002573041 (12.00 vs 10.00) (Line #1374)
01:54:39 am: Error: PokerStars: [2] Invalid pot size for hand #2021072002573043 (27.00 vs 12.00) (Line #1449)
01:54:39 am: Error: PokerStars: Invalid pot size (97.00 vs pot: 83.00 rake: 0.00 jpt: 0.00) for hand #2021072002573045 (Line #1544)
01:54:39 am: Error: PokerStars: [2] Invalid pot size for hand #2021072002573047 (269.00 vs 12.00) (Line #1629)
01:54:39 am: Error: PokerStars: Encountered unexpected end of hand history. (Hand #2021072002573048) (Line #1645)
01:54:39 am: Error: PokerStars: [2] Invalid pot size for hand #2021072002573049 (4.00 vs 2.00) (Line #1714)
01:54:39 am: Error: PokerStars: Encountered unexpected end of hand history. (Hand #2021072002573057) (Line #2042)
01:54:39 am: Error: PokerStars: Invalid pot size (0.00 vs pot: 24.00 rake: 0.00 jpt: 0.00) for hand #2021072002573058 (Line #2116)
01:54:39 am: Error: PokerStars: Encountered unexpected end of hand history. (Hand #2021072002573064) (Line #2306)
01:54:39 am: Error: PokerStars: [2] Invalid pot size for hand #2021072002573065 (32.00 vs 30.00) (Line #2379)
01:54:39 am: Error: PokerStars: Encountered unexpected end of hand history. (Hand #2021072002573073) (Line #2717)
01:54:39 am: Error: PokerStars: [2] Invalid pot size for hand #2021072002573074 (4.00 vs 2.00) (Line #2785)
01:54:39 am: Error: PokerStars: Encountered unexpected end of hand history. (Hand #2021072002573081) (Line #3037)
01:54:39 am: Error: PokerStars: [2] Invalid pot size for hand #2021072002573082 (12.00 vs 10.00) (Line #3088)
01:54:39 am: Error: PokerStars: [2] Invalid pot size for hand #2021072002573084 (12.00 vs 4.00) (Line #3152)
01:54:39 am: Error: PokerStars: Encountered unexpected end of hand history. (Hand #2021072002573088) (Line #3280)
01:54:39 am: Error: PokerStars: Invalid pot size (0.00 vs pot: 30.00 rake: 0.00 jpt: 0.00) for hand #2021072002573089 (Line #3345)
01:54:39 am: Error: PokerStars: ZayZay action out of sequence (Hand #2021072002573090) (Line #3386)
01:54:39 am: Error: PokerStars: [2] Invalid pot size for hand #2021072002573093 (53.00 vs 30.00) (Line #3502)
01:54:39 am: Error: PokerStars: Encountered unexpected end of hand history. (Hand #2021072002573095) (Line #3553)
01:54:39 am: Error: PokerStars: [2] Invalid pot size for hand #2021072002573096 (52.00 vs 50.00) (Line #3624)
01:54:39 am: Error: PokerStars: [2] Invalid pot size for hand #2021072002573098 (100.00 vs 90.00) (Line #3711)
01:54:39 am: Error: PokerStars: Encountered unexpected end of hand history. (Hand #20210720025730103) (Line #3891)
01:54:39 am: Error: PokerStars: Invalid pot size (845.00 vs pot: 843.00 rake: 0.00 jpt: 0.00) for hand #20210720025730104 (Line #3963)
01:54:39 am: Error: PokerStars: [2] Invalid pot size for hand #20210720025730105 (12.00 vs 4.00) (Line #3995)
01:54:39 am: Error: PokerStars: Encountered unexpected end of hand history. (Hand #20210720025730110) (Line #4157)
01:54:39 am: Error: PokerStars: [2] Invalid pot size for hand #20210720025730111 (12.00 vs 10.00) (Line #4214)
01:54:39 am: Error: PokerStars: Invalid pot size (0.00 vs pot: 12.00 rake: 0.00 jpt: 0.00) for hand #20210720025730112 (Line #4250)
01:54:39 am: Error: PokerStars: Encountered unexpected end of hand history. (Hand #20210720025730117) (Line #4423)
01:54:39 am: Error: PokerStars: [2] Invalid pot size for hand #20210720025730118 (4.00 vs 2.00) (Line #4480)
01:54:39 am: Error: PokerStars: Kanye2020 action out of sequence (Hand #20210720025730119) (Line #4524)
01:54:39 am: Error: PokerStars: [2] Invalid pot size for hand #20210720025730120 (12.00 vs 4.00) (Line #4557)
01:54:39 am: Error: PokerStars: Encountered unexpected end of hand history. (Hand #20210720025730124) (Line #4696)
01:54:39 am: Error: PokerStars: [2] Invalid pot size for hand #20210720025730125 (47.00 vs 45.00) (Line #4751)
01:54:39 am: Error: PokerStars: Encountered unexpected end of hand history. (Hand #20210720025730130) (Line #4907)
01:54:39 am: Error: PokerStars: Invalid pot size (179.00 vs pot: 177.00 rake: 0.00 jpt: 0.00) for hand #20210720025730131 (Line #4972)
01:54:39 am: Error: PokerStars: Encountered unexpected end of hand history. (Hand #20210720025730137) (Line #5162)
01:54:39 am: Error: PokerStars: Invalid pot size (74.00 vs pot: 72.00 rake: 0.00 jpt: 0.00) for hand #20210720025730138 (Line #5226)
01:54:39 am: Error: PokerStars: Encountered unexpected end of hand history. (Hand #20210720025730144) (Line #5444)
01:54:39 am: Error: PokerStars: [2] Invalid pot size for hand #20210720025730145 (10.00 vs 0.00) (Line #5509)
01:54:39 am: Error: PokerStars: [2] Invalid pot size for hand #20210720025730149 (49.00 vs 35.00) (Line #5663)
01:54:39 am: Error: PokerStars: Encountered unexpected end of hand history. (Hand #20210720025730151) (Line #5723)
01:54:39 am: Error: PokerStars: [2] Invalid pot size for hand #20210720025730152 (12.00 vs 10.00) (Line #5776)
01:54:39 am: Error: PokerStars: Encountered unexpected end of hand history. (Hand #20210720025730158) (Line #5987)
01:54:39 am: Error: PokerStars: [2] Invalid pot size for hand #20210720025730159 (12.00 vs 10.00) (Line #6041)
01:54:39 am: Error: PokerStars: [2] Invalid pot size for hand #20210720025730162 (45.00 vs 12.00) (Line #6159)
01:54:39 am: Error: PokerStars: Encountered unexpected end of hand history. (Hand #20210720025730165) (Line #6261)
01:54:39 am: Error: PokerStars: [2] Invalid pot size for hand #20210720025730166 (12.00 vs 10.00) (Line #6322)
01:54:39 am: Error: PokerStars: Victor Li action out of sequence (Hand #20210720025730168) (Line #6398)
01:54:39 am: Error: PokerStars: Encountered unexpected end of hand history. (Hand #20210720025730172) (Line #6538)
01:54:39 am: Error: PokerStars: [2] Invalid pot size for hand #20210720025730173 (56.00 vs 39.00) (Line #6600)
01:54:39 am: Error: PokerStars: Encountered unexpected end of hand history. (Hand #20210720025730179) (Line #6809)
01:54:39 am: Error: PokerStars: [2] Invalid pot size for hand #20210720025730180 (12.00 vs 10.00) (Line #6876)
01:54:39 am: Error: PokerStars: Invalid pot size (838.00 vs pot: 808.00 rake: 0.00 jpt: 0.00) for hand #20210720025730183 (Line #7008)
01:54:39 am: Error: PokerStars: Invalid pot size (55.00 vs pot: 35.00 rake: 0.00 jpt: 0.00) for hand #20210720025730184 (Line #7043)
01:54:39 am: Error: PokerStars: Encountered unexpected end of hand history. (Hand #20210720025730186) (Line #7090)
01:54:39 am: Error: PokerStars: Invalid pot size (47.00 vs pot: 45.00 rake: 0.00 jpt: 0.00) for hand #20210720025730187 (Line #7156)
01:54:39 am: Error: PokerStars: Encountered unexpected end of hand history. (Hand #20210720025730193) (Line #7379)
01:54:39 am: Error: PokerStars: [2] Invalid pot size for hand #20210720025730194 (4.00 vs 2.00) (Line #7442)
01:54:39 am: Error: PokerStars: Encountered unexpected end of hand history. (Hand #20210720025730199) (Line #7590)
01:54:39 am: Error: PokerStars: [2] Invalid pot size for hand #20210720025730200 (12.00 vs 10.00) (Line #7650)
01:54:39 am: Error: PokerStars: Encountered unexpected end of hand history. (Hand #20210720025730205) (Line #7819)
01:54:39 am: Error: PokerStars: [2] Invalid pot size for hand #20210720025730206 (141.00 vs 10.00) (Line #7871)
01:54:39 am: Error: PokerStars: Encountered unexpected end of hand history. (Hand #20210720025730210) (Line #7995)
01:54:39 am: Error: PokerStars: [2] Invalid pot size for hand #20210720025730211 (66.00 vs 64.00) (Line #8043)
01:54:39 am: Error: PokerStars: Encountered unexpected end of hand history. (Hand #20210720025730214) (Line #8112)
01:54:39 am: Error: PokerStars: Invalid pot size (112.00 vs pot: 110.00 rake: 0.00 jpt: 0.00) for hand #20210720025730215 (Line #8155)
01:54:39 am: Error: PokerStars: Encountered unexpected end of hand history. (Hand #20210720025730218) (Line #8212)
01:54:39 am: Error: PokerStars: [2] Invalid pot size for hand #20210720025730219 (12.00 vs 10.00) (Line #8255)
01:54:39 am: Error: PokerStars: Encountered unexpected end of hand history. (Hand #20210720025730222) (Line #8313)
01:54:39 am: Error: PokerStars: [2] Invalid pot size for hand #20210720025730223 (12.00 vs 10.00) (Line #8361)
01:54:39 am: Error: PokerStars: [2] Invalid pot size for hand #20210720025730224 (27.00 vs 12.00) (Line #8385)
01:54:39 am: Error: PokerStars: Encountered unexpected end of hand history. (Hand #20210720025730226) (Line #8419)
01:54:39 am: Error: PokerStars: [2] Invalid pot size for hand #20210720025730227 (33.00 vs 10.00) (Line #8467)
01:54:39 am: Error: PokerStars: Kanye2020 action out of sequence (Hand #20210720025730228) (Line #8495)
01:54:39 am: Error: PokerStars: Encountered unexpected end of hand history. (Hand #20210720025730230) (Line #8529)
01:54:39 am: Error: PokerStars: [2] Invalid pot size for hand #20210720025730231 (12.00 vs 10.00) (Line #8565)
01:54:39 am: Error: PokerStars: [2] Invalid pot size for hand #20210720025730233 (939.00 vs 67.00) (Line #8632)
01:54:39 am: Error: PokerStars: Encountered unexpected end of hand history. (Hand #20210720025730234) (Line #8643)
01:54:39 am: Error: PokerStars: [2] Invalid pot size for hand #20210720025730235 (12.00 vs 10.00) (Line #8678)
01:54:39 am: Error: PokerStars: Encountered unexpected end of hand history. (Hand #20210720025730240) (Line #8843)
01:54:39 am: Error: PokerStars: [2] Invalid pot size for hand #20210720025730241 (12.00 vs 10.00) (Line #8893)
01:54:39 am: Error: PokerStars: [2] Invalid pot size for hand #20210720025730245 (154.00 vs 64.00) (Line #9049)
01:54:39 am: Error: PokerStars: Encountered unexpected end of hand history. (Hand #20210720025730247) (Line #9110)
01:54:39 am: Error: PokerStars: [2] Invalid pot size for hand #20210720025730248 (35.00 vs 10.00) (Line #9177)
01:54:39 am: Error: PokerStars: [2] Invalid pot size for hand #20210720025730252 (12.00 vs 4.00) (Line #9316)
01:54:39 am: Error: PokerStars: Encountered unexpected end of hand history. (Hand #20210720025730254) (Line #9364)
01:54:39 am: Error: PokerStars: [2] Invalid pot size for hand #20210720025730255 (12.00 vs 10.00) (Line #9414)
01:54:39 am: Error: PokerStars: [2] Invalid pot size for hand #20210720025730257 (31.00 vs 12.00) (Line #9493)
01:54:40 am: Adjusting session stakes ...
01:54:41 am: Updating the cache ...
01:54:41 am: 		+ Complete (235 hands, 0 summaries, 98 errors, 0 duplicates)
01:54:41 am: Import complete. 524 hands in 3 files were imported. (120 errors, 100 duplicates)

**Comment by evolutionsoftswiss** [comment]
Okay, there were still issues in several hands according the PT log. I still think, it is not a NLH issue, but problems with unexpected characters in names. Here the NLH players used both spaces and $ in their names, causing the code to fail or produce incorrect output. Next try:
[handhistory-converter-0.2.4-program.zip](https://github.com/evolutionsoftswiss/pokernow-handhistory-converter/files/6899474/handhistory-converter-0.2.4-program.zip)

**Comment by benstuart0** [comment]
It all finally works! Great job, thanks for working this out with me.

**Comment by evolutionsoftswiss** [comment]
Thanks for the feedback, I can now publish an improved release.

**Comment by evolutionsoftswiss** [comment]
Fixed in latest Release v0.2.1
